### DISCUSSION TITLE
Session not saved between app/mounts

### DISCUSSION THREAD

**Discussion by okelet** [AnswerDotAI/fasthtml#610] [Q&A]
Hi, I am trying to create a minimal app that mounts another app. I have created this sample code:

```python
from fasthtml.common import A, FastHTML, Mount, P, fast_app, serve, uri, Titled  # pylint: disable=no-name-in-module


mount_app, mount_rt = fast_app()


@mount_rt("/")
def home(session: dict):
    """
    Home page.
    """
    return Titled(
        "Mount demo",
        P(f"Session value is {session.get('foo')}"),
    )


app: FastHTML
app, rt = fast_app(
    debug=True,
    live=True,
    secret_key="xxxxxxx",
    routes=[
        Mount("/subapp", mount_app, name="subapp"),
    ],
)


@rt("/set", name="set_session")
def set_session(session):
    """
    Set session.
    """
    session.setdefault("foo", 0)
    session["foo"] += 1
    return Titled("Set", P(f"Session set to {session.get('foo')}; click here to ", A("Get", link=uri("home"))))


@rt("/", name="home")
def get_session(session):
    """
    Get session.
    """
    session.setdefault("foo", 0)
    return Titled("Get", P(f"Session: {session.get('foo')}; click here to ", A("Set", link=uri("set_session")), "; go to ", A("subapp", link=uri("subapp:home"))))


serve()
```

When I access http://127.0.0.1:5001, the cookie is not set, and default value is 0. If I click on "Set" to change its value, and click back to "Get", I get the new value (1). But if click to go to /subapp, the session value is missing. How can I make the session to be shared between the main app and the mounted apps?

**Comment by esteban-ams** [comment]
I'm having a similar issue with sessions. When its called inside a function to modify it, works like a charm. But once the request its finished, when calling another endpoint via htmx, the session values are already gone.

What is the correct behavior of the sessions ?
